subsys/fs/fs.c

```diff
@@ -499,7 +499,9 @@ int fs_mkdir(const char *abs_path)
 	}
 
 	rc = mp->fs->mkdir(mp, abs_path);
-	if (rc < 0) {
+	if (rc == -EEXIST) {
+		LOG_ERR("directory already exists (%d)", rc);
+	} else if (rc < 0) {
```

Comment:
This is pointless, because the returned code already states the problem, so leave the `if (rc < 0)`.

Reply:
I was just thinking a more informative error message would be appropriate in the case of already existing directory. If I look at the logging output and see this instead of "failed to create directory", the information I receive about the error will be much more helpful. I can revert if you want though.

Reply:
What I mean is that is pointless to have specific message for that. You got the error code (`%d`) already anyway, and the app should actually post some info or do something according to the error code.

I have already written it in the https://github.com/zephyrproject-rtos/zephyr/issues/35718 that I, in general, agree that there is too much logging happening if VFS. Adding more specific logs does not fix the problem.

Reply:
@emillindq Can you remove the additional log?

Reply:
@emillindq Hi, could you follow the https://github.com/zephyrproject-rtos/zephyr/pull/36618/files#r665161118 comment? I would like to follow with approval and maybe the PR could get merged soon.

Reply:
@emillindq ping?

Reply:
@de-nordic Many apologies!! For some reason I missed the notifications on your comments, and I've had vacations for the past three weeks as well :/ Anyhow, I have fixed it now!

Reply:
> 
> 
> @de-nordic Many apologies!! For some reason I missed the notifications on your comments, and I've had vacations for the past three weeks as well :/ Anyhow, I have fixed it now!

@emillindq No problem, you do not have to apology.

Do you want to close the #35718, after the PR gets merged, or want to keep it for further approach to logging in VFS?

Reply:
@de-nordic We can close it; as long as it's possible to create a file/dir irregardless of its existence without triggering LOG_ERR I'm very happy :partying_face: